app/src/main/java/com/example/unlibrary/exchange/ExchangeViewModel.java

```diff
@@ -41,6 +72,47 @@ public ExchangeViewModel() {
         return this.mBooks;
     }
 
+    /**
+     * Getter for mCurrentBook live data object
+     *
+     * @return LiveData<Book> This returns the mBooks object
+     */
+    public LiveData<Book> getCurrentBook() {
+        return this.mCurrentBook;
+    }
+
+    /**
+     * Generates and saves the request into firestore.
+     */
+    public void sendRequest() {
+        Request request = new Request(mExchangeRepository.getUid(), mCurrentBook.getValue().getId());
+        mExchangeRepository.createRequest(request,
+                o -> {
+                    mNavigationEvent.setValue(ExchangeBookDetailsFragmentDirections.actionExchangeBookDetailsFragmentToExchangeFragment());
+                },
+                e -> {
+                    mFailureMsgEvent.setValue("Failed to send request.");
+                    Log.e(TAG, "Failed to send request.", e);
+                });
+    }
+
+    /**
+     * Sets mCurrentBook and navigates to detailed book view.
+     *
+     * @param view     view to navigate from.
+     * @param position list position of selected book.
+     */
+    public void selectCurrentBook(View view, int position) {
+        if (mBooks.getValue() == null) {
+            mFailureMsgEvent.setValue("Failed show details for book");
+        } else {
+            Book book = mBooks.getValue().get(position);
+            mCurrentBook.setValue(book);
+            NavDirections direction = ExchangeFragmentDirections.actionExchangeFragmentToExchangeBookDetailsFragment();
+            Navigation.findNavController(view).navigate(direction);
+        }
+    }

```

Comment:
Nit I personally like early returns, but feel free to dismiss this if you feel like its not important :smile: 
```suggestion
    public void selectCurrentBook(View view, int position) {
        if (mBooks.getValue() == null) {
            mFailureMsgEvent.setValue("Failed show details for book");
            return;
        } 
        
            Book book = mBooks.getValue().get(position);
            mCurrentBook.setValue(book);
            NavDirections direction = ExchangeFragmentDirections.actionExchangeFragmentToExchangeBookDetailsFragment();
            Navigation.findNavController(view).navigate(direction);
        
    }

```